src/cascadia/TerminalApp/Profile.cpp

```diff
@@ -1188,3 +1188,31 @@ GUID Profile::GetGuidOrGenerateForJson(const Json::Value& json) noexcept
 
     return Profile::_GenerateGuidForProfile(name, source);
 }
+
+// Method Description:
+// - Validates that the `colorScheme` set for this Profile exists. If it does
+//   exist, returns true. Otherwise, we'll initialize our color table to the
+//   default color scheme, the Campbell scheme, and returns false.
+// - This is done
+//   for microsoft/terminal#2547. In the case where the scheme was invalid, we'd
+//   leave the table un-initialized, resulting in all the colors being RGB(0, 0,
+//   0,) _black_.
+// Arguments:
+// - schemes: a list of color schemes to check and ensure that our colorScheme exitsts in.
+// Return Value:
+// - true if the color scheme existed, false otherwise.
+bool Profile::ValidateColorScheme(const std::vector<::TerminalApp::ColorScheme>& schemes) noexcept
+{
+    const ColorScheme* const matchingScheme = _FindScheme(schemes, _schemeName.value());

```

Comment:
curious why there's a loose vector of schemes, and a global function to search it for a named value. is there a better abstraction here?

Reply:
So yes, a map would make more sense here. 

Right now, we use `GlobalAppSettings::GetColorSchemes()` in a few places:
  1. tests
  2. enumerating each scheme to serialize them (unused currently)
  3. finding a scheme to possibly layer on it
  4. Creating a `TerminalSettings`
  5. _ValidateAllSchemesExist (added here)

2 is trivial to remove. 3 would likely be easy to replace with a `map`, since _yea we're looking it up by name_. 4 and 5 both use this _FindScheme function, which is also easy to remove. 1 is the scenario that'd be annoying to change, there's simply lots of places in the tests where we're going `_globals.GetColorSchemes().at(N)` to find the N'th scheme and check some value. Though I suppose that's not the _worst_ thing to change, and it is overall goodness.

Yea lets do it.